### PR TITLE
WINDUP-680: Java API Incorrectly Processes Interfaces & WINDUP-681: J…

### PR DESCRIPTION
**Pull request by jsight**
…ava API Should Provide Marker for Type: Interface, Class

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/900/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/269/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/902/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/271/
Test PASSed.

File: java-ast/tests/src/test/java/org/jboss/windup/ast/java/test/JavaASTReferenceResolverTest.java [hunk 1]
```diff
@@ -161,4 +161,18 @@ public void testNonImportedFQCN()
...
     }
+
+    @Test
+    public void testInterfaceExtension()
+    {
+        List<ClassReference> references = ASTProcessor.analyze(getLibraryPaths(), getSourcePaths(),
+                    Paths.get("src/test/resources/testclasses/simple/EventServer.java"));
+        for (ClassReference reference : references)
+        {
+            System.out.println("Reference: " + reference);
```
Comment (mbriskar) [inline java-ast/tests/src/test/java/org/jboss/windup/ast/java/test/JavaASTReferenceResolverTest.java]: This is a debug println and should probably be removed.

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java [hunk 2]
```diff
@@ -81,12 +82,12 @@ public void perform(GraphRewrite event, EvaluationContext context, JavaClassFile
...
                     {
                         JavaClassModel interfaceModel = javaClassService.getOrCreatePhantom(interfaceName);
-                        javaClassModel.addImplements(interfaceModel);
+                        javaClassService.addInterface(javaClassModel, interfaceModel);
                     }
                 }
 
                 String superclassName = bcelJavaClass.getSuperclassName();
-                if (!StringUtils.isBlank(superclassName))
+                if (!bcelJavaClass.isInterface() && !StringUtils.isBlank(superclassName))
```
Comment (mbriskar) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java]: What happened if the interface was let in? Is the superInterface registered now?

File: java-ast/tests/src/test/java/org/jboss/windup/ast/java/test/JavaASTReferenceResolverTest.java (hunk 1 above)
Comment (lincolnthree) [inline java-ast/tests/src/test/java/org/jboss/windup/ast/java/test/JavaASTReferenceResolverTest.java]: +1

File: rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java (hunk 2 above)
Comment (jsight) [inline rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/operation/AddClassFileMetadata.java]: The super-interface is just added to the list of "interfaces" for the JavaClassModel. This is simliar to how it is handled with java.lang.Class.

File: java-ast/tests/src/test/java/org/jboss/windup/ast/java/test/JavaASTReferenceResolverTest.java (hunk 1 above)
Comment (jsight) [inline java-ast/tests/src/test/java/org/jboss/windup/ast/java/test/JavaASTReferenceResolverTest.java]: I think these are useful for debugging tests, but I'll remove them anyway :)

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/915/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/285/
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/916/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/286/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/919/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/289/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/289/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/289/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/921/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/291/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-pr-builder/933/
Test PASSed.

**Comment by windupgithubbot1** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://10.3.8.5//job/windup-core-PR-dependents-check/303/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='org.jboss.windup.rules:windup-rulesets' /><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/303/org.jboss.windup.rules$windup-rulesets/testReport'>org.jboss.windup.rules:windup-rulesets</a>: <span class='status-failure'>1</span></h3><ul><li><a href='http://10.3.8.5/job/windup-core-PR-dependents-check/303/org.jboss.windup.rules$windup-rulesets/testReport/org.jboss.windup.rules.tests/WindupRulesTest/testWindupRules/'><strong>org.jboss.windup.rules.tests.WindupRulesTest.testWindupRules</strong></a></li></ul>
Test FAILed.

**Pull request merged**